source/MaterialXView/ShaderValidators/Glsl/GlslValidator.h

```diff
@@ -0,0 +1,326 @@
+#ifndef MATERIALX_GLSLVALIDATOR_H
+#define MATERIALX_GLSLVALIDATOR_H
+
+#include <MaterialXShaderGen/HwShader.h>
+#include <MaterialXView/ImageHandler.h>
+#include <vector>
+#include <string>
+#include <unordered_map>
+
+namespace MaterialX
+{
+
+/// Error string list type
+using ShaderValidationErrorList = std::vector<std::string>;
+
+/// @class @GlslValidator
+/// Helper class to perform validation of GLSL source code generated by the GLSL code generator.
+///
+/// There are two main interfaces which can be used. One which takes in a HwShader and one which
+/// allows for explicit setting of shader stage code.
+///
+/// The main services provided are:
+///     - Validation: All shader stages are compiled and atteched to a GLSL shader program.
+///     - Introspection: The compiled shader program is examined for uniforms and attributes.
+///     - Binding: Uniforms and attributes which match the predefined variables generated the GLSL code generator
+///       will have values assigned to this. This includes matrices, attribute streams, and textures.
+///     - Rendering: The program with bound inputs will be used to drawing geometry to an offscreen buffer.
+///     An interface is provided to save this offscreen buffer to disk using an externally defined image handler.
+///
+class GlslValidator
+{
+  public:
+    /// Structure to hold information about program inputs
+    struct ProgramInput
+    {
+        static int INVALID_OPENGL_TYPE;
+
+        /// Program location. -1 means an invalid location
+        int _location;
+        /// OpenGL type of the input. -1 means an invalid type
+        int _type;
+        /// Size.
+        int _size;
+
+        /// Program input constructor
+        ProgramInput(int inputLocation, int inputType, int inputSize)
+        : _location(inputLocation)
+        , _type(inputType)
+        , _size(inputSize)
+        {}
+    };
+    /// Program input structure shared pointer type
+    using ProgramInputPtr = std::shared_ptr<ProgramInput>;
+    /// Program input shaded pointer map type
+    using ProgramInputMap = std::unordered_map<std::string, ProgramInputPtr>;
+
+    /// Constructor
+    GlslValidator();
+
+    /// Destructor
+    virtual ~GlslValidator();
+
+    /// @name Setup
+    /// @{
+
+    /// Internal initialization of stages and OpenGL contstructs
+    /// required for program validation and rendering.
+    /// An exception is thrown on failure.
+    /// The exception will contain a list of initialization errors.
+    void initialize();
+
+    /// Return whether the validator has been initialized properly.
+    /// It is possible due to platform ahd machine restrictions that this
+    /// may fail.
+    bool iniitialized() const
```

Comment:
Type: double "ii"

Reply:
Since exception is thrown if initialize() fails, is this query method needed? I think removing it would make it more clear that exceptions are being used.